airflow/bin/cli.py

```diff
@@ -423,14 +418,11 @@ def unpause(args, dag=None):
 def set_is_paused(is_paused, args, dag=None):
     dag = dag or get_dag(args)
 
-    session = settings.Session()
-    dm = session.query(DagModel).filter(
-        DagModel.dag_id == dag.dag_id).first()
-    dm.is_paused = is_paused
-    session.commit()
```

Comment:
Do we want to remove all the `.commit()` calls too? That seems like without it changes won't be persisted back to the DB until later?

Reply:
So we should not remove any `.commit()` without consideration, but after the `create_session`, or `@provide_session` the session is committed anyway. Therefore the explicit `.commit()` in the code will only act as an in-between commit. We should close the session if we don't use it again directly.

Reply:
Here in the code: https://github.com/apache/incubator-airflow/blob/master/airflow/utils/db.py#L45

Reply:
Isn't the `commit` on https://github.com/apache/incubator-airflow/blob/master/airflow/utils/db.py#L45 after `expunge_all` which would remove all the object instance from the session?

Reply:
Wouldn't the commit on https://github.com/apache/incubator-airflow/blob/master/airflow/utils/db.py#L45 after expunge_all which would remove all the object instance from the session?

Reply:
I think the changes will still be committed to the database, otherwise the `expunge_all()` would have the same behavior as the `rollback()`.

Reply:
I am not sure about this - https://stackoverflow.com/a/44465147/5691525

Reply:
I've restored the `.commit()` for now. I would like to work this Friday on setting the `expire_on_commit=True`: https://github.com/apache/airflow/blob/master/airflow/settings.py#L198

It feels like we have a lot of connections to the database because they aren't properly closed.